src/index.ts

```diff
@@ -43,12 +46,13 @@ export default async function GleeAppInitializer () {
     showFunctionsDir: GLEE_FUNCTIONS_DIR !== resolve(GLEE_DIR, 'functions'),
   })
 
-  await registerFunctions(GLEE_FUNCTIONS_DIR)
-  await registerLifecycleEvents(GLEE_LIFECYCLE_DIR)
-
   const parsedAsyncAPI = await getParsedAsyncAPI()
   const channelNames = parsedAsyncAPI.channelNames()
 
+  await generateFunctions(parsedAsyncAPI, GLEE_DIR)
```

Comment:
Why did you choose to generate "empty" functions from the AsyncAPI file instead of interpreting the values directly? 🤔

Reply:
I don't get it. 🤔

Reply:
I was expecting that during the [Function `register` process](https://github.com/asyncapi/glee/blob/master/src/lib/functions.ts#LL54), you would read the AsyncAPI document searching for this kind of remote functions (those with URL) and would register them there. Instead, you're creating empty functions with an `invoke` property: https://github.com/asyncapi/glee/pull/372/files#diff-6f611215e3d2978b5127e7c36198526b54c97440b27b9fb25742c9a7ebfa0d0bR144.

I'm not saying one is better than the other, just curious why you took that route.

Reply:
I would say registering them in the `register` function would be much faster since there is no I/O involved, plus there is no conversion between URLs and file names. 🤦 
let me try that...